Build types file to enable access to enums

bruno12mota opened the pull request:
Hey! These changes enable the consumption of wp-types enums without requiring additional build on the consumption side. It doesn't change any behavior on the types.

```
import { WP_Post_Status_Name } from 'wp-types';

function getPosts(status: WP_Post_Status_Name = WP_Post_Status_Name.publish) {...}
```

Not only am i consuming WP_Post_Status_Name as a type but also as a value, so i think both should be possible.

bruno12mota committed:
Wp types build types file to enable access to enums

johnbillion commented:
Thanks for the PR! Do you have some docs on exactly what swc is doing as a result? Cheers.

bruno12mota commented:
@johnbillion swc in this case is just a faster alternative to tsc. so it's just transpiling the ts file to js. The consumable parts of the wp-types will be just the enums, here's an extract of one of the enums generated in that file.

```
export var WP_Post_Type_Name = /*#__PURE__*/ function(WP_Post_Type_Name) {
    WP_Post_Type_Name["post"] = "post";
    WP_Post_Type_Name["page"] = "page";
    WP_Post_Type_Name["attachment"] = "attachment";
    WP_Post_Type_Name["revision"] = "revision";
    WP_Post_Type_Name["nav_menu_item"] = "nav_menu_item";
    WP_Post_Type_Name["custom_css"] = "custom_css";
    WP_Post_Type_Name["customize_changeset"] = "customize_changeset";
    WP_Post_Type_Name["oembed_cache"] = "oembed_cache";
    WP_Post_Type_Name["user_request"] = "user_request";
    WP_Post_Type_Name["wp_block"] = "wp_block";
    WP_Post_Type_Name["wp_global_styles"] = "wp_global_styles";
    WP_Post_Type_Name["wp_navigation"] = "wp_navigation";
    WP_Post_Type_Name["wp_template"] = "wp_template";
    WP_Post_Type_Name["wp_template_part"] = "wp_template_part";
    return WP_Post_Type_Name;
}({});
```

Can add an entry on the readme with something like the following to illustrate better what this enables

```
const api: Promise<WP_REST_API_Posts> = wp.apiFetch( {
	path: '/wp/v2/posts/',
        queryParams: {
               status: WP_Post_Status_Name.publish
        }
} );
```

johnbillion commented:
Thanks. There's some docs on this topic here that would be good to add to: https://github.com/johnbillion/wp-json-schemas/tree/trunk/packages/wp-types#can-i-use-the-enums-as-values-in-my-code

bruno12mota committed:
Change documentation for enums consumption without relying on typescript

bruno12mota commented:
Changed that section a little, removed what was before a requirement and added one more example to illustrate importing a type and value (enum) separately. 

This also removes the requirement of transpiling the package when consuming on a project with for example NextJs where you'd need to set it as a package that needs tranpsilation (https://nextjs.org/docs/app/api-reference/config/next-config-js/transpilePackages). Don't think it's worth to put on documentation though since it's an expected behavior

bruno12mota commented:
any plans in merging this soon @johnbillion? or any changes required?

johnbillion commented:
Yep I'll get back to this ASAP
